services/points/points.service.js

```diff
@@ -0,0 +1,125 @@
+const { EmbedBuilder } = require('discord.js');
+const axios = require('axios');
+const config = require('../../config');
+
+axios.default.defaults.headers.get.Authorization = `Token ${config.pointsbot.token}`;
+
+class PointsService {
+  static API_URL = 'https://www.theodinproject.com/api/points';
+
+  static async handleInteraction(interaction) {
+    if (interaction.options.getSubcommand() === 'user') {
+      PointsService.displayUserPoints(interaction);
+    } else if (interaction.options.getSubcommand() === 'leaderboard') {
+      PointsService.displayLeaderboard(interaction);
+    } else {
+      PointsService.displayInfo(interaction);
+    }
+  }
+
+  static async displayInfo(interaction) {
+    const pointsEmbed = new EmbedBuilder()
+      .setColor('#cc9543')
+      .setTitle('Points in the TOP Discord server')
+      .setDescription(`
+Want to give credit where it's due? Show your appreciation for helpful members in our server by giving them a point! Mention their \`@name\` and add \`++\` or \`:star:\`
+
+**Example:**
+
+\`@username ++\`
+\`@username :star:\`
+
+**Club 40:**
+
+Users who have accumulated 40 points will be awarded a special role in recognition of their consistent helpfulness.
+For further details about our roles, please refer to [our discord #roles channel](https://discord.com/channels/505093832157691914/936424264180060200)
+
+**Point Inflation:**
+
+Please don't beg for points or abuse the point system.
+We will have a strict moderation policy in place which will include losing access to the bot or participation in the points system.
+
+Our goal is to maintain a positive and supportive community, where help and contributions are valued.
+      `);
+
+    await interaction.reply({ embeds: [pointsEmbed] });
+  }
+
+  static async displayUserPoints(interaction) {
+    const user = interaction.options.getUser('name');
+    const response = await axios.get(PointsService.API_URL);
+
+    // eslint-disable-next-line
+    const data = response.data.filter((member) => interaction.guild.members.cache.get(member.discord_id));
+
```

Comment:
I'm not sure why we're filtering for users that exist in the server. Won't this make the ranks in the leaderboard flaky? Since if a person with points leaves the server, it would change the leaderboard rank for everyone below them. I might have missed something and this could be the intended behavior. Please let me know your thoughts, thanks!